src/store/activeChain.ts

```diff
@@ -0,0 +1,13 @@
+import { create } from "zustand"
+
+import { enabledNetworks } from "@/components/WagmiProvider"
+
+export interface ActiveChainStore {
+  chainId: number
+  setChainId: (_chainId: number) => void
+}
+
+export const useActiveChain = create<ActiveChainStore>((set, _get) => ({
```

Comment:
[Zustand](https://github.com/pmndrs/zustand) is an awesome little library for global state management. Pretty much everything you need from Redux, with none of the boilerplate. 

If you find yourself needing global state, look at using Zustand instead of writing a context provider.